### PR TITLE
feat: concatenate attachments with same names

### PR DESCRIPTION
**Pull request by alexbrdn**

### CODE REVIEW

**Comment by enricoschuetz** [comment]
I have several points that I would like to put up for discussion:

1.  given: 
    + readme.md on TM-ID 1
    + readme.md on TM-ID 2
    + no readme.md on TM-NAME

     when: fetch readme.md on TM-NAME with --concat
     then: attachment not found
     
     -> does it make sense to allow a fetch on e.g. readme.md --concat, even TM-NAME has no readme.md ?
     
2.  `tmc attachment fetch ...`  -> flag `-o` like  tmc fetch ?

3.  Renaming an attachment by a command ?
    Currently you have to fetch the old, import it with a new name and delete the old one, right ?

File: README.md [hunk 1]
```diff
@@ -1,105 +1,3 @@
-# Thing Model Catalog CLI
+# V2 readme
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply README.md:1]: oopsie!

File: api/tm-catalog.openapi.yaml [hunk 2]
```diff
@@ -498,6 +498,12 @@ paths:
         - $ref: '#/components/parameters/TMName'
         - $ref: '#/components/parameters/AttachmentFileName'
         - $ref: '#/components/parameters/RepoDisambiguator'
+        - name: concat
+          in: query
+          description: Fetch a concatenation of the attachment to a TM name and homonymous attachments to all versions of the same
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply api/tm-catalog.openapi.yaml:503]: Agree. That would be more standard English.

File: cmd/attachment/attachment_import.go [hunk 3]
```diff
@@ -44,5 +45,6 @@ func init() {
 	attachmentCmd.AddCommand(attachmentImportCmd)
 	cmd.AddRepoDisambiguatorFlags(attachmentImportCmd)
 	attachmentImportCmd.Flags().StringP("media-type", "m", "", "Media type of the attachment. Guessed automatically, if the flag is not set.")
+	attachmentImportCmd.Flags().StringP("name", "n", "", "Use this name for the attachment instead of the original file's name")
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply cmd/attachment/attachment_import.go:48]: agreed

**Comment by alexbrdn** [comment]
> I have several points that I would like to put up for discussion:
> 
> 1. given:
>    
>    * readme.md on TM-ID 1
>    * readme.md on TM-ID 2
>    * no readme.md on TM-NAME
>    
>    when: fetch readme.md on TM-NAME with --concat
>    then: attachment not found
>    -> does it make sense to allow a fetch on e.g. readme.md --concat, even TM-NAME has no readme.md ?
> 2. `tmc attachment fetch ...`  -> flag `-o` like  tmc fetch ?
> 3. Renaming an attachment by a command ?
>    Currently you have to fetch the old, import it with a new name and delete the old one, right ?

As discussed, implemented 1. and 2.

File: internal/app/cli/attachment.go [hunk 4]
```diff
@@ -107,13 +108,31 @@ func AttachmentDelete(ctx context.Context, spec model.RepoSpec, tmNameOrId, atta
...
+	if f != nil && !f.IsDir() {
+		Stderrf("output target folder --output is not a folder")
+		return errors.New("output target folder --output is not a folder")
+	}
+	fullFilename := filepath.Join(outputPath, attachmentName)
+	err = os.WriteFile(fullFilename, content, 0660)
+	if err != nil {
+		Stderrf("could not write attachment to file %s: %v", fullFilename, err)
+		return err
+	}
```
Comment (enricoschuetz) [inline internal/app/cli/attachment.go:135]: Somehow the output path of of the `-o` flag is not automatically created if it does not yet exist.  
I get an error e.g. `could not write attachment to file .fetchtest\readme.md: open .fetchtest\readme.md: The system cannot find the path specified.`  (tested on windows)

Note that for fetching a ThingModel the output path is automatically created if it does not yet exist.

Comment (alexbrdn) [reply internal/app/cli/attachment.go]: fixed

Comment (enricoschuetz) [reply internal/app/cli/attachment.go]: The output path is now created, also if it does not yet exists.  

But there is another problem:

If I fetch an attachment that does not exists, I get correctly the message e.g.:
`Failed to fetch attachment readme1.md to nexus-x/meteocontrol/bluelog-xc/v1.0.120-20241002095710-2d7451305ad4.tm.json: attachment not found`
But there is an empty file written to the `-o` output path, in the example above an empty `readme1.md` is written.

Comment (alexbrdn) [reply internal/app/cli/attachment.go]: turns out this problem has previously existed, but it used to show itself only in the incorrect rc=0 when attachment cannot be fetched :-)
fixed now

**Pull request merged**